specification.md

```diff
@@ -551,37 +553,64 @@ from a `QUEUED` state to a `COMPLETED` state. However, implementations
 can introduce a synthetic `ACTIVE` state change.
 
 
-<a name="job-waitfor"></a>
+<a name="job-wait"></a>
+<a name="job-wait_timeinterval-jobstate@"></a>
 ```java
-JobStatus waitFor(TimeInterval? timeout, JobState targetStates...)
+JobStatus wait(TimeInterval? timeout, JobState targetStates...)
+    throws UnreachableStateException
 ```
 
 Waits until the job reaches either of the `targetStates` or until an
 amount of time indicated by the timeout parameter passes. Returns the
 [JobStatus](#jobstatus) object that has one of the desired `targetStates`
-or `null` if the timeout is reached.
+or `null` if the timeout is reached. If none of the states in
+`targetStates` can be reached (such as, for example, because the job has
+entered the `FAILED` state while `targetStates` consists of `COMPLETED`),
+this method throws an
+[`UnreachableStateException`](#unreachablestateexception).
+
+<div class="imp-note">
+Implementations are encouraged to throw the `UnreachableStateException`
+as soon as it can be determined that the `targetStates` are unreachable
+and not necessarily when the job reaches a final state. However, whether
+it is possible to make such a determination before a final state is
+reached depends on the precise state model and on the exact time when
```

Comment:
The state model is precise, in the sense that we define it in this document - so I would suggest to remove that part.

Reply:
Are you sure? I meant it as a future-proof thing, in case we will need to update the state model. Our current state model is very carefully engineered to make this unnecessary, but I worry that it may end up insufficient.

Reply:
If we change the state model we probably have to change text in several places.  As it is, I would avoid the impression that the state model is flexible or exchangeable, or that the semantics somehow depends on the underlying batch system state model...